internal/logger/telemetry_store_watermill.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2024 Stacklok, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logger
+
+import (
+	"fmt"
+
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/rs/zerolog"
+
+	"github.com/stacklok/minder/internal/engine"
+	minderv1 "github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1"
+)
+
+// TelemetryStoreWMMiddleware is a Watermill middleware that
+// logs the relevant telemetry data.
+type TelemetryStoreWMMiddleware struct {
+	l *zerolog.Logger
+}
```

Comment:
If we store a logger here, does that override [our usage of `zerowater.NewZerologLoggerAdapter`](https://github.com/stacklok/minder/blob/main/internal/events/eventer.go#L126)?

Reply:
that adapted is only usable if you have access to the watermill router. Which we don't here. I think a good thing to do would be to make the eventer also receive a zerolog Logger instance, so they both can use the same level at least.